packages/protocol/contracts/L1/libs/LibTokenomics.sol

```diff
@@ -17,32 +17,67 @@ import {TaikoToken} from "../TaikoToken.sol";
 
 library LibTokenomics {
     using LibMath for uint256;
-    uint256 private constant SZABO_TO_WEI = 1E12;
+    uint256 private constant TWEI_TO_WEI = 1E12;
+
+    error L1_INSUFFICIENT_TOKEN();
+    error L1_INVALID_PARAM();
 
     function withdraw(
         TaikoData.State storage state,
-        AddressResolver resolver
+        AddressResolver resolver,
+        uint256 amount
     ) internal {
         uint256 balance = state.balances[msg.sender];
-        if (balance <= 1) return;
+        if (balance <= amount) revert L1_INSUFFICIENT_TOKEN();
+
+        uint256 x;
+        {
+            x = balance - amount;
+        }
```

Comment:
Missing `unchecked`? I think this function should also be used quite infrequently so maybe not worth it to complicate the code to save some gas?